aladdin-container.sh

```diff
@@ -161,6 +168,61 @@ function _add_authentication_user_to_kubeconfig() {
 EOT
 }
 
+function _handle_aws_config() {
+    # This function does quite a few things with aws credentials/config files:
+    # 1) Move the mounted aws creds to ~/.aws
+    # 2) Checks if we have the BASTION_ACCOUNT_ENABLED configuration from our aladdin-config
+    # 3) If we do:
+    #        - export appropriate BASTION_ variables for other aladdin commands
+    #        - sanity test our aws configuration for the bastion account if INIT and not IS_LOCAL
+    #        - appropriately update our credentials/config files with the desired possible roles
+    #            - here we are calling the add-aws-assume-role-config aladdin command
+    # 4) If we do not:
+    #        - sanity test our aws configuration for the cluster's current aws account if INIT and not IS_LOCAL
+    # 5) Export AWS_PROFILE=$AWS_DEFAULT_PROFILE because kops uses the former
+
+    # Move aws credentials away from mount so we don't edit the host's aws files
+    cp -r /root/tmp/.aws /root/.aws
+    # See if bastion account is enabled
+    export BASTION_ACCOUNT_ENABLED="$(_extract_cluster_config_value bastion_account_enabled)"
+    if "$BASTION_ACCOUNT_ENABLED"; then
+        # Export BASTION_ACCOUNT_PROFILE, needed by add-aws-assume-role-config command
+        export BASTION_ACCOUNT_PROFILE="$(_extract_cluster_config_value bastion_account_profile)"
+        # Need to unset AWS_DEFAULT_PROFILE because aws requires that entry to exist even if we
+        # are specifying --profile to a separate present entry, which is expected in the bastion case
+        local aws_default_profile_temp="$AWS_DEFAULT_PROFILE"
+        unset AWS_DEFAULT_PROFILE
+        # Test aws config for bastion account if INIT and not IS_LOCAL
+        if "$INIT" && ! "$IS_LOCAL"; then
+            _test_aws_config "$BASTION_ACCOUNT_PROFILE"
+        fi
+        # Alias the add assume role config command
+        add_assume_role_config="$ALADDIN_DIR/commands/bash/container/add-aws-assume-role-config/add-aws-assume-role-config"
+        # Need to add aws configuration based on publish configuration.
+        # We need to do this because the publish ECR may be a different aws account than the one
+        # your cluster is provisioned in
+        publish_config="$(_extract_cluster_config_value publish)"
```

Comment:
It seems in some places we use the following config pattern:
`prefix_setting_name` (ex. bastion_account_profile)
while in others we put it inside a dictionary and extract using `jq`.

Might be good to unify on a single pattern at some point, not specifically related to this PR though

Reply:
Yeah, I was frustrated with the configuration extraction in bash. The reason some are inner nested and some aren't is due to the cluster override configs, specifically that, currently with the way the _extract_cluster_config function is written, they only override at the top level. So if we wanted to override just one part of the inner dictionary in a cluster specific config.json, we currently need to move that entire inner dictionary to the top level.

I can make a ticket to clean up config extraction in the bash part of aladdin

Reply:
got it, makes sense, merging nested configs does sound hairy

Reply:
https://fivestars.atlassian.net/projects/ALDN/issues/ALDN-148